public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -384,6 +386,37 @@ export class LokiDatasource
     return res.data || [];
   }
 
+  async queryStatsRequest(query: LokiQuery): Promise<QueryStats> {
+    const { start, end } = this.getTimeRangeParams();
+
+    const labelMatcherPositions = getStreamSelectorPositions(query.expr);
+    const labelMatchers = labelMatcherPositions.map((labelMatcher) => {
+      return query.expr.slice(labelMatcher.from, labelMatcher.to);
```

Comment:
I'd make utility function out of this. The same way as we have `getLogQueryFromMetricsQuery` in https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/queryUtils.ts, we could have `getStreamSelectorFromQuery`, so it could be reused in the future.